bentoml/_internal/frameworks/xgboost.py

```diff
@@ -162,120 +123,104 @@ def save(
-    name: t.Optional[str] = None,
-) -> "_XgBoostRunner":
     """
-    Runner represents a unit of serving logic that can be scaled horizontally to
-    maximize throughput. `bentoml.xgboost.load_runner` implements a Runner class that
-    wrap around a Xgboost booster model, which optimize it for the BentoML runtime.
+    Save a model instance to BentoML modelstore.
 
     Args:
-        tag (:code:`Union[str, Tag]`):
-            Tag of a saved model in BentoML local modelstore.
-        predict_fn_name (:code:`str`, default to :code:`predict`):
-            Options for inference functions. If you want to use `run`
-            or `run_batch` in a thread context then use `inplace_predict`.
-            Otherwise, `predict` are the de facto functions.
-        booster_params (:code:`t.Dict[str, t.Union[str, int]]`, default to :code:`None`):
-            Parameters for boosters. Refers to `Parameters docs <https://xgboost.readthedocs.io/en/latest/parameter.html>`_ for more information.
+        name (:code:`str`):
+            Name for given model instance. This should pass Python identifier check.
+        model (`xgboost.core.Booster`):
+            Instance of model to be saved
+        labels (:code:`Dict[str, str]`, `optional`, default to :code:`None`):
+            user-defined labels for managing models, e.g. team=nlp, stage=dev
+        custom_objects (:code:`Dict[str, Any]]`, `optional`, default to :code:`None`):
+            user-defined additional python objects to be saved alongside the model,
+            e.g. a tokenizer instance, preprocessor function, model configuration json
+        metadata (:code:`Dict[str, Any]`, `optional`, default to :code:`None`):
+            Custom metadata for given model.
+        model_store (:mod:`~bentoml._internal.models.store.ModelStore`, default to :mod:`BentoMLContainer.model_store`):
+            BentoML modelstore, provided by DI Container.
 
     Returns:
-        :obj:`~bentoml._internal.runner.Runner`: Runner instances for :mod:`bentoml.xgboost` model
+        :obj:`~bentoml.Tag`: A :obj:`tag` with a format `name:version` where `name` is the user-defined model's name, and a generated `version` by BentoML.
 
     Examples:
 
     .. code-block:: python
 
-        import xgboost as xgb
-        import bentoml.xgboost
-        import pandas as pd
+    """
+
 
-        input_data = pd.from_csv("/path/to/csv")
-        runner = bentoml.xgboost.load_runner("my_model:20201012_DE43A2")
-        runner.run(xgb.DMatrix(input_data))
+def get_runnable(
+    bento_model: bentoml.Model,
+) -> t.Type[bentoml.Runnable]:
     """
-    return _XgBoostRunner(
-        tag=tag,
-        predict_fn_name=predict_fn_name,
-        booster_params=booster_params,
-        name=name,
-    )
+    Private API: use :obj:`~bentoml.Model.to_runnable` instead.
+    """
+
+    class XGBoostRunnable(bentoml.Runnable):
+        SUPPORT_NVIDIA_GPU = True
+        SUPPORT_CPU_MULTI_THREADING = True
```

Comment:
I think we should do something like

```python
SUPPORTED_RESOURCES = set(bentoml.resources.NVIDIA_GPU, bentoml.resources.MULTIPLE_THREADS)
```

or maybe

```python
SUPPORTED_RESOURCES = [bentoml.resources.NVIDIA_GPU]
SUPPORTS_CPU_MULTI_THREADING = True
```

so that this is more generic.

Reply:
Users should have the flexibility to support resources not listed under `bentoml.resources` with a custom strategy. So we might just keep them as strings.